### DISCUSSION TITLE
cmake error: clang is not able to compile a simple test program

### DISCUSSION THREAD

**Discussion by chenghuaWang** [android/ndk#2022] [Q&A] [answered]
I'm new to NDK. And I am trying to compile a lib using ndk toolchain. The cmake file I used is shown below:

```cmake
cmake .. \
-DCMAKE_TOOLCHAIN_FILE=$ANDROID_NDK/build/cmake/android.toolchain.cmake \
-DCMAKE_BUILD_TYPE=Release \
-DANDROID_ABI="arm64-v8a" \
-DANDROID_NATIVE_API_LEVEL=android-28  \
-DNATIVE_LIBRARY_OUTPUT=. -DNATIVE_INCLUDE_OUTPUT=. $1 $2 $3 \
-DDEBUG=ON \
-DTEST=OFF \
-DARM=ON \
-DAPK=OFF
```

And cmake gave error:

```shell
-- The C compiler identification is unknown
-- The CXX compiler identification is unknown
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - failed
-- Check for working C compiler: /media/wch/D/commonlib/android-ndk-r26d/toolchains/llvm/prebuilt/linux-x86_64/bin/clang
-- Check for working C compiler: /media/wch/D/commonlib/android-ndk-r26d/toolchains/llvm/prebuilt/linux-x86_64/bin/clang - broken
CMake Error at /home/wch/Downloads/cmake-3.28.1-linux-x86_64/share/cmake-3.28/Modules/CMakeTestCCompiler.cmake:67 (message):
  The C compiler

    "/media/wch/D/commonlib/android-ndk-r26d/toolchains/llvm/prebuilt/linux-x86_64/bin/clang"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: '/media/wch/D/Project/mllm/build-arm/CMakeFiles/CMakeScratch/TryCompile-UICPj5'
    
    Run Build Command(s): /home/wch/Downloads/cmake-3.28.1-linux-x86_64/bin/cmake -E env VERBOSE=1 /usr/bin/gmake -f Makefile cmTC_07689/fast
    /usr/bin/gmake  -f CMakeFiles/cmTC_07689.dir/build.make CMakeFiles/cmTC_07689.dir/build
    gmake[1]: 进入目录“/media/wch/D/Project/mllm/build-arm/CMakeFiles/CMakeScratch/TryCompile-UICPj5”
    Building C object CMakeFiles/cmTC_07689.dir/testCCompiler.c.o
    /media/wch/D/commonlib/android-ndk-r26d/toolchains/llvm/prebuilt/linux-x86_64/bin/clang   -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security   -o CMakeFiles/cmTC_07689.dir/testCCompiler.c.o -c /media/wch/D/Project/mllm/build-arm/CMakeFiles/CMakeScratch/TryCompile-UICPj5/testCCompiler.c
    clang-17: error: no input files
    gmake[1]: *** [CMakeFiles/cmTC_07689.dir/build.make:78：CMakeFiles/cmTC_07689.dir/testCCompiler.c.o] error 1
    gmake[1]: 离开目录“/media/wch/D/Project/mllm/build-arm/CMakeFiles/CMakeScratch/TryCompile-UICPj5”
    gmake: *** [Makefile:127：cmTC_07689/fast] error 2
    
    

  

  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:3 (project)
```

I'm confused that why there has a `clang-17: error: no input files` error.

**Comment by DanAlbert** [comment]
It's very unlikely that anyone will be able to help without a repro case.

**Comment by Fleeym** [comment]
Hey there, I'm also seeing this behaviour. Same error (no input files), the compiler identification being "unknown". 
I've only set `ANDROID_ABI` and `ANDROID_PLATFORM` as env vars. My guess is that it could be related to my system configuration?

I'm using Fedora 40 (KDE), with the latest clang installed (shouldn't matter), cmake v3.28.2, and ndk r26d

**Reply by DanAlbert** [reply]
It's very unlikely that anyone will be able to help without a repro case.

**Reply by Fleeym** [reply]
I don't really know what could be considered a "repro case" for this. I just can't use the ndk under any circumstance. The only repro I could give is my system configuration. If I could offer any specific info, I'd be glad to help!

**Reply by DanAlbert** [reply]
Code for the failing build and instructions to build it. There's a description of what we need on the bug form: https://github.com/android/ndk/issues/new?assignees=&labels=bug&projects=&template=bug-report.yml&title=%5BBUG%5D

If you think it's a problem with your system configuration, docker is invaluable for that kind of thing.

**Reply by Fleeym** [reply, marked as answer]
After some mindless digging I seem to have fixed it. The problem was caused by the ndk being somehow corrupted by my unzip software (ark). Used `unzip` on the archive and it now works. Weird.

**Reply by OAguinagalde** [reply]
I can confirm that `expand-archive` (powershell command) gave me the same problem. `unzip` did work. I'm assuming that this is probably related to the `deferred symbolic links` that `unzip` does and that I imagine `expand-archive` doesn't.
